src/net-ipfs-engine/Catalyst.Ipfs.Http.Server/HttpApi/V0/ConfigController.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Catalyst.Ipfs.Core.CoreApi;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Catalyst.Ipfs.Http.Server.HttpApi.V0
+{
+
+    /// <summary>
+    ///   Manages the IPFS Configuration.
+    /// </summary>
+    public class ConfigController : IpfsController
+    {
+        /// <summary>
+        ///   Details on a configuration setting.
+        /// </summary>
+        public class ConfigDetailDto
+        {
+            /// <summary>
+            ///   The name of the configuration setting.
+            /// </summary>
+            public string Key;
+
+            /// <summary>
+            ///   The value of the configuration setting.
+            /// </summary>
+            public JToken Value;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make this field 'private' and encapsulate it in a 'public' property.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4612924)